File: tests/datetime/data/datetime.cpp
using EStyle = icu_72::DateFormat::EStyle;

typedef std::tuple<UDate, EStyle, EStyle, const char*> testCase;
typedef std::vector<const char*> langids;
typedef std::vector<testCase> values;
typedef std::tuple<langids, values> localeTestData;
typedef std::vector<localeTestData> testData;

testData data = {
  localeTestData(
    {"en"},
    {
      testCase(27832853, EStyle::NONE, EStyle::NONE, ""),
      testCase(15932853, EStyle::NONE, EStyle::SHORT, "11:33 AM"),
      testCase(18932853, EStyle::NONE, EStyle::MEDIUM, "7:33:00 PM"),
      testCase(18932853, EStyle::SHORT, EStyle::NONE, "12/30/05"),
      testCase(16282853, EStyle::MEDIUM, EStyle::NONE, "Dec 16, 2000"),
      testCase(19012153, EStyle::SHORT, EStyle::MEDIUM, "2/23/06, 9:13:00 PM"),
      testCase(33102153, EStyle::MEDIUM, EStyle::SHORT, "Dec 8, 2032, 2:33 PM"),
      testCase(39100853, EStyle::SHORT, EStyle::SHORT, "5/5/44, 8:53 AM"),
      testCase(31923823, EStyle::MEDIUM, EStyle::MEDIUM, "Sep 12, 2030, 7:43:00 AM"),
      testCase(27114532, EStyle::LONG, EStyle::MEDIUM, "July 21, 2021 at 12:52:00 PM"),
    }
  ),
};

File: tests/datetime/data/datetime.rs
pub fn get_data() -> TestData<'static> {
    TestData(
        Box::new([
            LocaleTestData {
                langid: Box::new(["en"]),
                values: Box::new([
                    TestCase {
                        input: 27832853,
                        style: ("None", "None"),
                        output: "",
                    },
                    TestCase {
                        input: 15932853,
                        style: ("None", "Short"),
                        output: "11:33 AM",
                    },
                    TestCase {
                        input: 18932853,
                        style: ("None", "Medium"),
                        output: "7:33:00 PM",
                    },
                    TestCase {
                        input: 18932853,
                        style: ("Short", "None"),
                        output: "12/30/05",
                    },
                    TestCase {
                        input: 16282853,
                        style: ("Medium", "None"),
                        output: "Dec 16, 2000",
                    },
                    TestCase {
                        input: 19012153,
                        style: ("Short", "Medium"),
                        output: "2/23/06, 9:13:00 PM",
                    },
                    TestCase {
                        input: 33102153,
                        style: ("Medium", "Short"),
                        output: "Dec 8, 2032, 2:33 PM",
                    },
                    TestCase {
                        input: 39100853,
                        style: ("Short", "Short"),
                        output: "5/5/44, 8:53 AM",
                    },
                    TestCase {
                        input: 31923823,
                        style: ("Medium", "Medium"),
                        output: "Sep 12, 2030, 7:43:00 AM",
                    },
                    TestCase {
                        input: 27114532,
                        style: ("Long", "Medium"),
                        output: "July 21, 2021 at 12:52:00 PM",
                    },
                ])
            },
        ])
    )
}

